com.unity.shadergraph/Editor/Data/Util/GraphUtil.cs

```diff
@@ -139,7 +139,15 @@ internal static bool CheckForRecursiveDependencyOnPendingSave(string saveFilePat
             {
                 foreach (var sgNode in subGraphNodes)
                 {
-                    if ((sgNode.asset.assetGuid == overwriteGUID) || sgNode.asset.descendents.Contains(overwriteGUID))
+                    var asset = sgNode?.asset;
+                    if (asset == null)
```

Comment:
From a maintenance perspective, at first glance I'd think this `if(asset == null) { // big comment } else { code }` block is a bug. I think putting the comment above `if(asset == null) return false;` is clearer in the intent.

Reply:
how about a continue; ?